docs/developers/SubstraitModifications.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: page
+title: Substrait Modifications
+nav_order: 4
+parent: Developer Overview
+---
+# Substrait Modifications in Gluten
+
+Substrait is a project aiming to create a well-defined, cross-language specification for data compute operations.
+Since it is still under active development, there are some lacking representations for Gluten needed computing
+operations. At the same time, some existing representations need to be modified a bit to satisfy the needs of computing.
+This page records all the Gluten changes to Substrait proto files for reference. It is preferred to upstream
+these changes to Substrait, but for those cannot be upstreamed, alternatives like `AdvancedExtension` could be
+considered.  
+
+## Modifications to algebra.proto
+
+* Added `JsonReadOptions` and `TextReadOptions` in `FileOrFiles`([#1584](https://github.com/oap-project/gluten/pull/1584)).
+* Changed join type `JOIN_TYPE_SEMI` to `JOIN_TYPE_LEFT_SEMI` and `JOIN_TYPE_RIGHT_SEMI`([#408](https://github.com/oap-project/gluten/pull/408)).
+* Added `WindowRel`, added `column_name` and `window_type` in `WindowFunction`,
+changed `Unbounded` in `WindowFunction` into `Unbounded_Preceding` and `Unbounded_Following`, and added WindowType([#485](https://github.com/oap-project/gluten/pull/485)).
+* Added `output_schema` in RelRoot([#1901](https://github.com/oap-project/gluten/pull/1901)).
+* Added `ExpandRel`([#519](https://github.com/oap-project/gluten/pull/519)).
```

Comment:
Please change the ExpandRel link to https://github.com/oap-project/gluten/pull/1361.

Reply:
Fixed.